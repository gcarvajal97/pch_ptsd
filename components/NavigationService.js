import { NavigationActions } from 'react-navigation';
import { DrawerActions } from 'react-navigation-drawer';

// Props is apparently to be deprecated soon, I've researched the current best practice and created this "service"
// This service will allow us to make calls to move screens using the navigate function below
// I also added in an openDrawer function (so that it can be called from the header button on MainScreen.js)
let _navigator;

function setTopLevelNavigator(navigatorRef) {
  _navigator = navigatorRef;
}

function getTopLevelNavigator() {
  return _navigator;
}

// Can be used to navigate to other screens, but currently Drawer only references the stack navigator
function navigateDrawer(routeName, params) {
  _navigator.dispatch(
    NavigationActions.navigate({
      routeName,
      params,
    })
  );
}

// Can be used to move into other screens defined in the stack navigator. 
function navigate(routeName, params) {
    _navigator.dispatch(
      NavigationActions.navigate({
        routeName,
        params,
      })
    );
  }

// Created this function to allow the openDrawer function to be called from the header in MainScreen.js
function openDrawer() {
    _navigator.dispatch(DrawerActions.openDrawer())
}

export default {
  navigate,
  navigateDrawer,
  setTopLevelNavigator,
  getTopLevelNavigator,
  openDrawer,
};