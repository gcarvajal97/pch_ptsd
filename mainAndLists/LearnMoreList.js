import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Card, ListItem, Divider } from 'react-native-elements';
import NavigationService from '../components/NavigationService';
import translate from '../components/translateService';
<<<<<<< HEAD
=======

// List of screens that will be displayed in this list component and their titles.
const listNames = [
    { name: translate('listLearnMore.notAlone'), navigateTo: 'NotAlone', accessibilityHint: translate('listLearnMore.notAloneHint')},
    { name: translate('listLearnMore.reactions'), navigateTo: 'Reactions', accessibilityHint: translate('listLearnMore.reactionsHint')},
    { name: translate('listLearnMore.stressReactions'), navigateTo: 'TraumaticStressReactions', accessibilityHint: translate('listLearnMore.stressReactionsHint')},
    { name: translate('listLearnMore.howLong'), navigateTo: 'HowLong', accessibilityHint: translate('listLearnMore.howLongHint')},
]
>>>>>>> cee48697c2d1bf7f32974783e959c3b9b9bf51f6

export default class LearnMoreList extends Component {
    /**
     * List component that renders the views and links to various aspects 
     * of the Learn More About Injury section of the application
     */

    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
<<<<<<< HEAD
        headerTitle: translate('learnMoreList.header')
=======
        headerTitle: translate('listLearnMore.header')
>>>>>>> cee48697c2d1bf7f32974783e959c3b9b9bf51f6
    }
    render() {
        return (
            <View>
                <Card containerStyle={{margin:5}}>
                    <Text style={{textAlign: 'center', backgroundColor:'#2089DC', fontSize:20, color:'white', padding:10, paddingVertical:10, margin:0}}>
<<<<<<< HEAD
                        {translate('learnMoreList.title')}
=======
                        {translate('listLearnMore.whatToExpect')}
>>>>>>> cee48697c2d1bf7f32974783e959c3b9b9bf51f6
                    </Text>
                    <Divider style={{backgroundColor:'black', height:1, marginBottom:-1}}/>
                    <View>
                        {
                            translate('learnMoreList.listNames').map((l, i) => {
                                return (
                                    <ListItem
                                        containerStyle={{paddingLeft:14, paddingVertical:12, paddingRight:0}}
                                        key={i}
                                        title={l.name}
                                        titleStyle={{fontSize:19, alignContent:'stretch', color:'black'}}
                                        onPress={()=>{NavigationService.navigate(l.navigateTo)}}
                                        accessible
                                        accessibilityLabel= {l.navigateTo}
                                        accessibilityHint= {l.accessibilityHint}
                                        chevron={{size:32, color: 'black'}}
                                        bottomDivider
                                    />
                                );
                            })
                        }
                    </View>
                </Card>
            </View>
        );
    }
}