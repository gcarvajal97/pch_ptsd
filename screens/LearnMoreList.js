import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { Card, ListItem, Divider } from 'react-native-elements';

import NavigationService from '../components/NavigationService';

// List of screens that will be dispalyed in this list component
const listNames = [
    { name: 'You Are Not Alone', navigateTo: 'NotAlone' },
    { name: 'Reactions to Injury', navigateTo: 'Reactions'},
    { name: 'What Are Traumatic Stress Reactions?', navigateTo: 'TSReactions'},
    { name: 'How Long Do Traumatic Stress Reactions Last?'},
]

export default class LearnMoreList extends Component {

    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
        headerTitle: 'Learn More'
    }
    render() {
        return (
            <View>
                <Card containerStyle={{margin:5}}>
                    <Text style={{textAlign: 'center', backgroundColor:'#2089DC', fontSize:20, color:'white', padding:10, paddingVertical:10, margin:0}}>
                        WHAT TO EXPECT AFTER INJURY
                    </Text>
                    <Divider style={{backgroundColor:'black', height:1, marginBottom:-1}}/>
                    <View>
                        {
                            listNames.map((l, i) => {
                                return (
                                    <ListItem
                                        containerStyle={{paddingLeft:14, paddingVertical:12, paddingRight:0}}
                                        key={i}
                                        title={l.name}
                                        titleStyle={{fontSize:19, alignContent:'stretch', color:'black'}}
                                        onPress={()=>{NavigationService.navigate(l.navigateTo)}}
                                        chevron={{size:32, color: 'black'}}
                                        bottomDivider
                                    />
                                );
                            })
                        }
                    </View>
                </Card>
            </View>
        );
    }
}