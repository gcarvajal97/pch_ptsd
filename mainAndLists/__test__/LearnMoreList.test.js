import React from 'react';
import LearnMoreList from '../LearnMoreList'
import renderer from 'react-test-renderer';
import Enzyme, { shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NavigationService from '../../components/NavigationService';
import { ListItem } from 'react-native-elements';

Enzyme.configure({ adapter: new Adapter() });
jest.mock('../../components/NavigationService')

describe("LearnMoreList", () => {

    afterEach(() => {
        jest.clearAllMocks();
      });

    it("matches Snapshot", () => {
        const snap = renderer.create(<LearnMoreList />).toJSON();
        expect(snap).toMatchSnapshot();
    });

    it("navigates to NotAlone page on button press", () => {
        const wrapper = shallow(<LearnMoreList />);
        wrapper.find(ListItem).at(0).props().onPress();
        expect(NavigationService.navigate).toHaveBeenCalledWith('NotAlone');
    });

    it("navigates to Reactions page on button press", () => {
        const wrapper = shallow(<LearnMoreList />);
        wrapper.find(ListItem).at(1).props().onPress();
        expect(NavigationService.navigate).toHaveBeenCalledWith('Reactions');
    });

    it("navigates to TraumaticStressReactions page on button press", () => {
        const wrapper = shallow(<LearnMoreList />);
        wrapper.find(ListItem).at(2).props().onPress();
        expect(NavigationService.navigate).toHaveBeenCalledWith('TraumaticStressReactions');
    });

    it("navigates to HowLong page on button press", () => {
        const wrapper = shallow(<LearnMoreList />);
        wrapper.find(ListItem).at(3).props().onPress();
        expect(NavigationService.navigate).toHaveBeenCalledWith('HowLong');
    });
});