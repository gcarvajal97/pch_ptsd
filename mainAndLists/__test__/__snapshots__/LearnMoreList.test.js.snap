// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnMoreList matches Snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "margin": 5,
        "marginBottom": 0,
        "padding": 15,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "#2089DC",
            "color": "white",
            "fontSize": 20,
            "margin": 0,
            "padding": 10,
            "paddingVertical": 10,
            "textAlign": "center",
          }
        }
      >
        WHAT TO EXPECT AFTER INJURY
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "black",
            "height": 1,
            "marginBottom": -1,
          }
        }
      />
      <View>
        <View
          accessibilityHint="Navigates to the you are not alone screen"
          accessibilityLabel="NotAlone"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 14,
                "paddingRight": 0,
                "paddingVertical": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignContent": "stretch",
                    "backgroundColor": "transparent",
                    "color": "black",
                    "fontSize": 19,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                You Are Not Alone
              </Text>
            </View>
            <View
              width={16}
            />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityHint="Navigates to the reactions to injury screen"
          accessibilityLabel="Reactions"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 14,
                "paddingRight": 0,
                "paddingVertical": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignContent": "stretch",
                    "backgroundColor": "transparent",
                    "color": "black",
                    "fontSize": 19,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Reactions To Injury
              </Text>
            </View>
            <View
              width={16}
            />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityHint="Navigates to the what are traumatic stress reactions screen"
          accessibilityLabel="TraumaticStressReactions"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 14,
                "paddingRight": 0,
                "paddingVertical": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignContent": "stretch",
                    "backgroundColor": "transparent",
                    "color": "black",
                    "fontSize": 19,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                What Are Traumatic Stress Reactions?
              </Text>
            </View>
            <View
              width={16}
            />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityHint="Navigates to the how long do traumatic stress reactions last screen"
          accessibilityLabel="HowLong"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
                "paddingLeft": 14,
                "paddingRight": 0,
                "paddingVertical": 12,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignContent": "stretch",
                    "backgroundColor": "transparent",
                    "color": "black",
                    "fontSize": 19,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                How Long Do Traumatic Stress Reactions Last?
              </Text>
            </View>
            <View
              width={16}
            />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
